### PR TITLE
Updated CRUD for User #15

### PR DESCRIPTION
**Pull request by nkomiljon**
Updated CRUD User to Applicant

### CODE REVIEW

**Review by IslomMakhsudov — DISMISSED** [review]

File: src/Application.Contracts/Applicant/Commands/UpdateApplicantCommand.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿using MediatR;
+
+namespace MRA.Jobs.Application.Contracts.Applicant.Commands;
+
+public class UpdateApplicantCommand : IRequest<Guid>
+{
+    public Guid Id { get; set; }
```
Comment (Muhammadislom99) [inline src/Application.Contracts/Applicant/Commands/UpdateApplicantCommand.cs:7]: дигар пропертихош чи ?

File: src/Application/Features/Applicant/Command/UpdateApplicant/UpdateApplicantCommandValidator.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace MRA.Jobs.Application.Features.Applicant.Command.UpdateApplicant;
+
+public class UpdateApplicantCommandValidator
+{
+
```
Comment (Muhammadislom99) [inline src/Application/Features/Applicant/Command/UpdateApplicant/UpdateApplicantCommandValidator.cs:5]: валидацияша нависед

File: src/Application.Contracts/Applicant/Commands/UpdateApplicantCommand.cs (hunk 1 above)
Comment (nkomiljon) [reply src/Application.Contracts/Applicant/Commands/UpdateApplicantCommand.cs:7]: Qush shud

File: src/Application/Features/Applicant/Command/UpdateApplicant/UpdateApplicantCommandValidator.cs (hunk 2 above)
Comment (nkomiljon) [reply src/Application/Features/Applicant/Command/UpdateApplicant/UpdateApplicantCommandValidator.cs]: Validatsiya navista shud

**Review by Muhammadislom99 — APPROVED** [review]

**Comment by nkomiljon** [comment]
Check plase Pullrequst

**Review by daler0022 — APPROVED** [review]

**Review by Muhammadislom99 — APPROVED** [review]

**Review by IslomMakhsudov — APPROVED** [review]
Good job!

File: src/Application/Features/Reviewer/Command/UpdateReviewer/UpdateReviewerCommandValidator.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using MRA.Jobs.Application.Contracts.Reviewer.Command;
+
+namespace MRA.Jobs.Application.Features.Reviewer.Command.UpdateReviewer;
+
+public class UpdateReviewerCommandValidator : AbstractValidator<UpdateReviewerCommand>
+{
+    public UpdateReviewerCommandValidator()
+    {
+        RuleFor(r => r.Id).NotEmpty();
```
Comment (mahmud-sharipov) [inline src/Application/Features/Reviewer/Command/UpdateReviewer/UpdateReviewerCommandValidator.cs:9]: add other properties to command and add valiation

File: src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using MRA.Jobs.Application.Contracts.Reviewer.Queries;
+using MRA.Jobs.Application.Contracts.Reviewer.Response;
+
+namespace MRA.Jobs.Application.Features.Reviewer.Query.GetAllReviewer;
+
+public class GetAllReviewerCommandHandler : IRequestHandler<GetAllReviewerQuery, ReviewerListDTO>
```
Comment (mahmud-sharipov) [inline src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandHandler.cs:6]: Not completed yet. Seems you did not finish this handler

File: src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandValidator.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using MRA.Jobs.Application.Contracts.Reviewer.Queries;
+
+namespace MRA.Jobs.Application.Features.Reviewer.Query.GetAllReviewer;
+
+public class GetAllReviewerCommandValidator : AbstractValidator<GetReviewerByIdQuery>
```
Comment (mahmud-sharipov) [inline src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandValidator.cs:5]: Remove this validator

**Review by mahmud-sharipov — COMMENTED** [review]
Please review and fix CR comments

File: src/Application/Features/Reviewer/Command/DeleteReviewer/DeleteReviewerCommandValidator.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿namespace MRA.Jobs.Application.Features.Reviewer.Command.DeleteReviewer;
+
+public class DeleteReviewerCommandValidator
```
[the remark this answers is not in the document]
Comment (nkomiljon) [reply src/Application/Features/Reviewer/Command/DeleteReviewer/DeleteReviewerCommandValidator.cs]: ok

File: src/Application/Features/Applicant/Query/GetApplicantById/GetApplicantQueryHandler.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    public async Task<ApplicantDetailsDto> Handle(GetApplicantByIdQuery request, CancellationToken cancellationToken)
+    {
+        var applicant =
+            await _context.Applicants.FirstOrDefaultAsync(a => a.Id == request.Id, cancellationToken: cancellationToken);
+
```
[the remark this answers is not in the document]
Comment (nkomiljon) [reply src/Application/Features/Applicant/Query/GetApplicantById/GetApplicantQueryHandler.cs]: ok

File: src/Application/Features/Reviewer/Command/UpdateReviewer/UpdateReviewerCommandValidator.cs (hunk 3 above)
Comment (nkomiljon) [reply src/Application/Features/Reviewer/Command/UpdateReviewer/UpdateReviewerCommandValidator.cs]: ok

File: src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandValidator.cs (hunk 5 above)
Comment (nkomiljon) [reply src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandValidator.cs]: ok

File: src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandHandler.cs (hunk 4 above)
Comment (nkomiljon) [reply src/Application/Features/Reviewer/Query/GetAllReviewer/GetAllReviewerCommandHandler.cs]: ok

File: src/Application.Contracts/Applicant/Queries/GetAllApplicantQuery.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+﻿using MediatR;
+using MRA.Jobs.Application.Contracts.Applicant.Responses;
+
+namespace MRA.Jobs.Application.Contracts.Applicant.Queries;
+
+public class GetAllApplicantQuery : IRequest<List<ApplicantListDto>>
+{
+  //  public Guid Id { get; set; }
```
Comment (IslomMakhsudov) [inline src/Application.Contracts/Applicant/Queries/GetAllApplicantQuery.cs:8]: Remove your comment

File: src/Application.Contracts/Applicant/Responses/ApplicantListDto.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
+﻿namespace MRA.Jobs.Application.Contracts.Applicant.Responses;
+
+public class ApplicantListDto
```
Comment (IslomMakhsudov) [inline src/Application.Contracts/Applicant/Responses/ApplicantListDto.cs:3]: Your List and Details Dto are same

Comment (IslomMakhsudov) [reply src/Application.Contracts/Applicant/Responses/ApplicantListDto.cs:3]: It shoud be different

File: src/Application.Contracts/Reviewer/Response/ReviewerListDTO.cs [hunk 10]
```diff
@@ -0,0 +1,27 @@
+﻿using MediatR;
+
+namespace MRA.Jobs.Application.Contracts.Reviewer.Response;
+
+public class ReviewerListDto
```
Comment (IslomMakhsudov) [inline src/Application.Contracts/Reviewer/Response/ReviewerListDTO.cs:5]: Fix it too

File: src/Application/Features/Applicant/Command/CreateApplicant/CreateApplicantCommandHandler.cs [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    private readonly IMapper _mapper;
+    private readonly IApplicationDbContext _context;
+
+    public CreateApplicantCommandHandler(
+        IApplicationDbContext context,
+        IMapper mapper)
+    {
+        _mapper = mapper;
+        _context = context;
+        /*_dataTime = dateTime;
```
Comment (IslomMakhsudov) [inline src/Application/Features/Applicant/Command/CreateApplicant/CreateApplicantCommandHandler.cs:17]: If you don't need this code don't comment it, Just delete it.

File: src/Application.Contracts/Applicant/Queries/GetAllApplicantQuery.cs (hunk 8 above)
Comment (nkomiljon) [reply src/Application.Contracts/Applicant/Queries/GetAllApplicantQuery.cs:8]: ok

File: src/Application.Contracts/Applicant/Responses/ApplicantListDto.cs (hunk 9 above)
Comment (nkomiljon) [reply src/Application.Contracts/Applicant/Responses/ApplicantListDto.cs:3]: ok

File: src/Application.Contracts/Reviewer/Response/ReviewerListDTO.cs (hunk 10 above)
Comment (nkomiljon) [reply src/Application.Contracts/Reviewer/Response/ReviewerListDTO.cs:5]: fixed

File: src/Application/Features/Applicant/Command/CreateApplicant/CreateApplicantCommandHandler.cs (hunk 11 above)
Comment (nkomiljon) [reply src/Application/Features/Applicant/Command/CreateApplicant/CreateApplicantCommandHandler.cs:17]: fixed

**Review by IslomMakhsudov — DISMISSED** [review]
Good job.

**Review by daler0022 — APPROVED** [review]

**Review by daler0022 — DISMISSED** [review]

**Review by IslomMakhsudov — APPROVED** [review]

**Review by daler0022 — APPROVED** [review]

**Pull request merged**